### PR TITLE
[#157006854] Admin (Caterer) should be able to manage meal options

### PR DESCRIPTION
**Pull request by SendiSteve**
#### What does this PR do?
Have CRUD on resources endpoint
#### Description of Task to be completed?
Have the following endpoints working for the meals

```
GET/v1/meals/
```
```
POST/v1/meals/
```

```
GET/v1/meals/<string:name>
```

```
PUT/v1/meals/<string:name>
```

```
DELETE/v1/meals/<string:name>
```
#### How should this be manually tested?
- After cloning the repo, cd into it and RUN python run.py
- Using Postman test every endpoint above with this header.
_key_: Content-Type _value_: application/json
#### What are the relevant pivotal tracker stories?
#157006854

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 app = Flask(__name__, instance_relative_config=True)
 
 api = Api(app)
 
 # Route
-api.add_resource(Hello, '/Hello')
 api.add_resource(MealListResource, '/v1/meals/', endpoint='meals')
-api.add_resource(MealResource, '/v1/meal/<string:name>', endpoint='meal')
\ No newline at end of file
+api.add_resource(MealResource, '/v1/meals/<string:name>', endpoint='meal')
```
Comment (Descartess) [inline app/__init__.py]: Great start!. However consider using ids to extract meal information rather than using names

File: models/meal.py [hunk 2]
```diff
@@ -1,4 +1,3 @@
-
 class Meal(object):
     def __init__(self, name, price, description):
         self.name = name
```
Comment (Descartess) [inline models/meal.py]: Consider having an id to uniquely identify a meal

File: resources/meal.py [hunk 3]
```diff
@@ -1,78 +1,51 @@
...
-                return {
-                    'message':
-                    "Meal with that name '{}' already created.".format(name)
-                    }
-        # data = MealResource.parser.parse_args()
-
-        meal = {'name': name}
-        meals.append(meal)
-        return meal, 201
+        data = request.get_json(force=True)
```
Comment (Descartess) [inline resources/meal.py]: You can get rid of `force=True` if you specify in postman that you are using `application/json`

File: tests/test_meal.py [hunk 4]
```diff
@@ -7,39 +7,82 @@ class MealsTestCase(unittest.TestCase):
...
+            'name': 'Matooke',
+            'price': 3445,
+            'description': 'very good'
+        }))
         self.assertEqual(res.status_code, 201)
         self.assertIn('Matooke', str(res.data))
-
+    
     def test_api_can_get_all_meals(self):
+        "Test API can get all meals in a list"
```
Comment (Descartess) [inline tests/test_meal.py]: This should be a doc string comment

Comment (SendiSteve) [reply tests/test_meal.py]: I have also corrected that

File: resources/meal.py (hunk 3 above)
Comment (SendiSteve) [reply resources/meal.py]: I have removed the force=True

**Pull request closed without merging**